From b7a117d9306fde4d77293f136771d3e08f237ff9 Mon Sep 17 00:00:00 2001
From: Eliot Lee <eliot.lee@intel.com>
Date: Tue, 10 Feb 2015 04:16:45 -0800
Subject: [PATCH 295/307] atomisp2: [ACC] Fixing WDT driver recovery with acc
 extensions.

This fixes the issue that ACC firmware configuration gets overwritten
by successive create_streams/load_acc_extensions calls.

Change-Id: I1c4dcf4a1eec7f4d5db5fa817c0b2c2126983c68
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-17167
Signed-off-by: Eliot Lee <eliot.lee@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/40950
Reviewed-by: Kriti Pachhandara <kriti.pachhandara@intel.com>
Reviewed-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Reviewed-by: Soren Friis <soren.friis@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c    | 4 ----
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c  | 2 ++
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 66f4e82..7d37077 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1332,10 +1332,6 @@ static void __atomisp_css_recover(struct atomisp_device *isp, bool isp_timeout)
 		if (!stream_restart[i])
 			continue;
 
-		if (atomisp_acc_load_extensions(asd) < 0)
-			dev_err(isp->dev,
-					"acc extension failed to reload\n");
-
 		if (isp->inputs[asd->input_curr].type != FILE_INPUT)
 			atomisp_css_input_set_mode(asd,
 					CSS_INPUT_MODE_SENSOR);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index 41f37ca..944d1ef 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -1279,6 +1279,8 @@ int atomisp_css_start(struct atomisp_sub_device *asd,
 			ret = -EINVAL;
 			goto stream_err;
 		}
+		/* in_reset == true, extension firmwares are reloaded after the recovery */
+		atomisp_acc_load_extensions(asd);
 	}
 
 	/*
-- 
1.9.1

